src/elm/Pages/Pipeline/View.elm

```diff
@@ -221,32 +222,55 @@ wrapPipelineConfigurationContent model cls content =
         body
 
 
-{-| viewTemplatesExpansion : takes model and renders the config header button for expanding pipeline templates.
+{-| viewPipelineActions : takes model and renders the config header buttons for expanding pipeline templates and downloading yaml.
 -}
-viewTemplatesExpansion : PartialModel a -> Html Msg
-viewTemplatesExpansion model =
-    case model.templates of
-        ( Success templates, _ ) ->
-            if Dict.size templates > 0 then
-                div [ class "expand-templates", Util.testAttribute "pipeline-templates-expand" ]
-                    [ expandTemplatesToggleIcon model.pipeline
-                    , expandTemplatesToggleButton model.pipeline
-                    , expandTemplatesTip
+viewPipelineActions : PartialModel a -> Html Msg
+viewPipelineActions model =
+    let
+        t =
+            case model.templates.data of
+                Success templates ->
+                    if Dict.size templates > 0 then
+                        div [ class "action", class "expand-templates", Util.testAttribute "pipeline-templates-expand" ]
+                            [ expandTemplatesToggleButton model.pipeline
+                            , expandTemplatesToggleIcon model.pipeline
+                            , expandTemplatesTip
+                            ]
+
+                    else
+                        text ""
+
+                _ ->
+                    text ""
+
+        d =
+            div [ class "action" ]
+                [ button
+                    [ class "button"
+                    , class "-link"
+                    , Util.testAttribute <| "download-yml"
+                    , onClick <| DownloadLogs velaYmlFileName <| RemoteData.unwrap "" .data <| Tuple.first model.pipeline.config
+                    , attribute "aria-label" <| "download pipeline configuration file for "
                     ]
+                    [ text <|
+                        if model.pipeline.expanded then
+                            "download (expanded) " ++ velaYmlFileName
 
-            else
-                text ""
+                        else
+                            "download " ++ velaYmlFileName
+                    ]
+                ]
+    in
+    div [ class "actions" ] [ t, d ]
 
-        ( Loading, _ ) ->
-            text ""
 
-        _ ->
-            text ""
+velaYmlFileName =
+    "vela.yml"
```

Comment:
A user could have a `.vela.yml` or `.vela.yaml` I don't think there is a clever way to know that. I personally think it's fine to just assume `.yml` or be opinionated and only download it to that file extension. 

But it might be an odd user experience to see `.yml` when the user stored their config in their repo as `.yaml`. Also, might not need to be fixed in this PR but might want to have a story to track fixing it.

Reply:
a story to track it, and adding that to the headers of the GET call might work.
but i TOTALLY agree, but as of now theres no way to tell which one theyre using.